genfond/execute_datalog_policy.py

```diff
@@ -0,0 +1,99 @@
+from .feature_generator import FeaturePool
+from .ground import ground
+from .state_space_generator import check_formula, apply_action_effects
+import itertools
+import logging
+import random
+
+log = logging.getLogger(__name__)
+
+
+def get_next_state(states):
+    return random.choice([state for state in states])
+
+
+def action_string(action) -> str:
+    return f'{action.name}({",".join([str(p) for p in action.parameters])})'
+
+
+def state_string(state) -> str:
+    return ",".join([str(p) for p in state])
+
+
+def execute_datalog_policy(domain, problem, datalog_policy, max_steps=0):
+    log.info(f'Executing policy:\n{datalog_policy}\nin {domain.name} for problem {problem.name}')
+
+    log.debug("Grounding actions...")
+    grounded_actions = ground(domain, problem)
+    log.debug("Grounding actions done.")
+    
+    features = []
+    for rule in datalog_policy.rules:
+        for concepts in rule.tail_by_parameter.values():
+            features.extend(concepts)
+    
+    featurePool = FeaturePool(domain, [problem], max_complexity=15, preset_features=features)
```

Comment:
As discussed, you shouldn't construct the feature pool, as this will not work for large instances. You need to construct the features manually instead (see `execute_policy.py`)